### PR TITLE
Remove link to “planet.apache.org” and update conference links.

### PR DESCRIPTION
**Pull request by rafaelweingartner**
Remove link to “planet.apache.org” and update conference links and update the link for the CCCs hot site

### CODE REVIEW

**Comment by terbolous** [comment]
The actual changes looks good, but shouldn't you update the markdown in `source/` rather than the generated html?

**Comment by swill** [comment]
Yes he should, I missed that. When you build it will update the html files, but the markdown/rst should be updated.

**Comment by rafaelweingartner** [comment]
@terbolous, @swill, 
do you mean changing the ".erb" file and then generating the HTML using the "build.sh" to generate the html files?

**Comment by pdion891** [comment]
Hi, I think we miss-managed our repo. The original idea was to have only source files in the master branch for the website, and when the PR is aprouve we build it in the asf-site branch and then include the html version of it, this is the branch that get pushed into webservers.

I might worth update the Readme about this :-S

**Comment by rafaelweingartner** [comment]
@pdion891, I also thought that, but then, I asked on Slack, and it was suggested for me to commit the “compiled/generated” HTML files. I thought that a Jenkins build was getting the PRs that were merged and automatically building the HTML files and then copying them to a web server. It seems this process is manual.

BTW: @swill, and @terbolus, I did not change the HTML files directly. I only changed the “source/layouts/layout.erb” file, and then generated the new HTML files using the “”build.sh” script.

**Comment by pdion891** [comment]
The website is not build via jenkins yet. manual for now :-S

**Comment by rafaelweingartner** [comment]
Do we have a jenkins? I could configure a job to do that. Then, we would not need to version HTML files.

**Comment by pdion891** [comment]
we have builds.cloudstack.org, but we would need to validate how to push files on webserver which I think the current case if a pull process.

**Comment by rafaelweingartner** [comment]
by validate you mean checking if everything is ok with the changes pushed to the webserver, right?

If we use a hook on merge at the "apache/cloudstack-www", can we assume that the content is validated? 
I mean, we will only merge PRs after reviews; it seems that we can consider changes introduced by a merge on "apache/cloudstack-www" validated.

**Comment by swill** [comment]
We can't merge into `apache/cloudstack-www`, that repo is read only and is only updated when we push to the upstream ASF repo, so that is too late in the process I think.  Also, the build server would have to have access to push to ASF since that is where the site is published from, but that is a problem because we have to track who has made changes to the ASF repo and that person has to have a valid ICLA.  I think we are stuck doing it in our environments for now.

**Comment by rafaelweingartner** [comment]
We cannot merge into "apache/cloudstack-www", but the merges on ASF repo are synchronized with "apache/cloudstack-www". Could not we create a Jenkins job that builds the website on every new commit on "apache/cloudstack-www"? The changes are trackable, I believe we are already tracking changes through commits on “apache/cloudstack-www” (by commits on “apache/cloudstack-www”, I mean on ASF repo that is then mirrored on “apache/cloudstack-www”).

That part of ICLA is already covered, not? For me, it feels that is the same process for ACSs source code. At the end, only committers can merge to ASF repos, and every committer has to have a signed ICLA.

We are diverging on the topic of the PR. Do you (all) think we should push it further, or maybe close it until we figure out the questions mentioned here?

**Comment by swill** [comment]
So the flow would be:
- pull from github/asf
- make changes
- build locally to verify the changes, but the resulting markup is not tracked (removed + ignored)
- push changes to asf (must be an ICLA committer)
- changes get sent to github
- jenkins builds the static site and pushes it to the asf `asf-site` branch (must be an ICLA committer)

Since we have not removed the ICLA committer from the flow, we are fine.  That is still a manual step.  But right now the static site is hosted from the ASF infra (my understanding), so the content has to be tracked in the ASF repo for the static site to be served.  So that means that even if we don't track the content in the master branch, we have to in the `asf-site` branch.  Jenkins has to push to that ASF repo to publish the site, so that has to be *someones* ICLA associated with that commit.

If the site was not hosted on ASF infra, I don't think we would have the problem because we would still be building and deploying from an ICLA approved version of the content.

Maybe I am reading too far into this, but I feel like we should understand the change before we do it.

I think we should go ahead with this commit since this is the flow we have been using, and take the jenkins discussion (which I think is awesome) to a different PR.

**Comment by rafaelweingartner** [comment]
Now I got your point. We do not have direct access to the webserver. Someone needs to commit these source files anyways. 
;)

**Comment by swill** [comment]
Right, so the site currently goes live with the ICLA commit, so why add more steps (kinda).

We could remove the built content from the master branch though.  I think we could probably do a `git rm` and then add them in `.gitignore` and the current flow would still work.  Maybe that is something we could consider.

**Comment by terbolous** [comment]
Actually @swill, searching through JIRA and infra mailing lists it seems that it is possible for a role commit by certain ASF jenkins slaves [0], [1]

Talked to @Humbedooh a bit about it on Slack today as well, and it seems doable - if we want to.

[0] https://issues.apache.org/jira/browse/INFRA-12425
[1] https://issues.apache.org/jira/browse/INFRA-9599

**Comment by swill** [comment]
OK. What would it be solving for?  Just trying to understand because we have to build the site locally anyway to check the changes and then we have to push the changes (which currently publishes the site).  Maybe I am missing something.

**Comment by rafaelweingartner** [comment]
After all that has been posted here, I am inclined to say that nothing would change.
The Jenkins would build/"compile" the HTML files, and then it has to commit to a repo that is synchronized with a web server, right?

Is the repo that provides the HTML   the cloudstack-www on ASF? Then nothing seems to change. We still keep versioning "compiled" HTML files and not just the source code. 

I might be missing some other details, though.

**Comment by rafaelweingartner** [comment]
I agree with @swill.

If you want, I could add these details in the README about the integration between the web server at ASF and this "cloudstack-www" repo.

**Comment by swill** [comment]
If you wouldn't mind, we might as well.  We also need to document that the
upstream ASF repo is at `
https://git-wip-us.apache.org/repos/asf/cloudstack-www.git`.  I don't think
that is documented anywhere, so committers won't know where to setup their
remotes as in order to push changes.  Thanks Rafael...  :)

On Fri, Feb 3, 2017 at 2:17 PM, Rafael Weingärtner <notifications@github.com

**Comment by rafaelweingartner** [comment]
Sure I do not mind.
I was assuming that the repo was going to be at https://git-wip-us.apache.org, but I added information in the README regarding this one.

Do you guys think we need to add/remove something?

**Comment by swill** [comment]
We may want to actually call out the user flow and commands to make the process easier for people.  This would be in addition to what you have already written.  Correct this if I am wrong about any of the following. 
```
# fork the repo to your own github: <your_user>
$ git clone https://github.com/apache/cloudstack-www.git
$ cd cloudstack-www
$ git remote add upstream https://git-wip-us.apache.org/repos/asf/cloudstack-www.git
$ git remote add <your_user> https://github.com/<your_user>/cloudstack-www.git
$ git checkout master
# make changes on master
$ middleman build
$ middleman server
# validate locally that the changes are good
$ ./build.sh
$ git status
$ git add .
$ git commit -am "your commit message"
$ git push <your_user> master
# create a PR against the github.com repo
# get feedback on the PR and proceed once PR review is accepted
$ git checkout asf-site
$ git merge master
$ git push upstream asf-site
# site will automatically be published live
# verify the changes on the live site.
```

This make sense?

**Comment by rafaelweingartner** [comment]
yeas, it makes.

I was assuming that any commiter that gets to here should be used to these steps; but, I will add a more detailed set of instructions like the one you proposed.

**Comment by swill** [comment]
Unless you know how everything works behind the scene (which is like nobody pretty much), then it can be hard to understand initially.  I think this makes it easier for more people to contribute to the documentation and the website.

It is only a committer who has to publish the change.  It may be a random person who is not a committer who creates the PR though.  That would be awesome.  :)

**Comment by rafaelweingartner** [comment]
I agree with you. The friendlier the better.

Sometimes we might think that those details are not important only because we have been around for so long here. On the other hand, a newcomer may feel lost without these details. I will add a more detailed step by step tonight.

**Comment by rafaelweingartner** [comment]
What about now?
I might have overlooked of missed some step. If you spot something inconsistent, please tell me.

**Comment by rafaelweingartner** [comment]
Any further thoughts here guys?
@terbolous, @swill, @pdion891

File: README.md [hunk 1]
```diff
@@ -15,14 +15,42 @@ Build the site with:
...
+
+# Commiting 
+Use GitHub PR to make changes. **No one should commit directly to Apache remote repositories without opening a PR and waiting for proper review**.
+If you got the feedback and have amended the changes (if needed), it is time to **commit** the changes to **Apache Software Foundation (AFS) remote repository**. The ASF remote repository for this project is: `https://git-wip-us.apache.org/repos/asf/cloudstack-www.git`
+
+**IMPORTANT:** To publish your changes, you should commit into the **asf-site** branch (pay attention, this is the name of the branch in the remote repository). **Do not forget**, you also have to commit the same changes to the master; so, **master** and **asf-site** branches are synchronized.
+
+The **asf-site** branch is synchronized with a web server that delivers the Apache CloudStack web pages.
+
+GOTCHA, the `build` directory needs to be renamed to `content` in the asf-site branch for the site to be served properly; If you use the `build.sh` script, this steep is already taken care of.
```
Comment (swill) [inline README.md]: s/steep/step/

File: README.md [hunk 2]
```diff
@@ -15,14 +15,42 @@ Build the site with:
...
+* `cd cloudstack-www`
+* __This step is only required for committers:__ (add the ASF remote repo) `git remote add upstream https://git-wip-us.apache.org/repos/asf/cloudstack-www.git`
+* if you have cloned  directly from Apache namespace on Github, then you need to add your namespace on Github as well, `git remote add <your_github_user> https://github.com/<your_github_user>/cloudstack-www.git`
+* (get the master branch): `git checkout -b master origin/master` or `git checkout -b master <your_github_user>/master`
+* Now, it is your turn to make the changes you want
+* `middleman build` will build/"compile" the HTML files from the sources
+* ` middleman server [-p <portnumber>]`. This will serve/deliver your HTML files over HTTP. You just gotta access them using a browser; e.g. `http://localhost:[port_configured]`. **Check your changes!!**
+* Execute `./build.sh`
+* `git add -A`
+* git commit -am "your commit message"
```
Comment (swill) [inline README.md]: `git commit -am "your commit message"` (put in backticks so it shows as a command)

File: README.md [hunk 3]
```diff
@@ -15,14 +15,42 @@ Build the site with:
...
+* git push `<your_github_user>` master
```
Comment (swill) [inline README.md]: `git push <your_github_user> master` (again in backticks so it formats nicely in readme)

File: README.md [hunk 4]
```diff
@@ -15,14 +15,42 @@ Build the site with:
...
+* Check if the changes appear properly on your Github project, and then create a PR against the Apache `cloudstack-www` repo.
+* Get feedback on the PR and proceed once PR review is accepted
+* If you are not a committer, your job finishes here. Congratulations you have made the Apache CloudStack website better  :thumbsup:
+* Continuing, for committers. Clone or add the repo of our contributor on Github using `git remote add <friend_contributor> https://github.com/<friend_contributor>/cloudstack-www.git`
+* git checkout asf-site
```
Comment (swill) [inline README.md]: `git checkout asf-site`  (added backticks)

File: README.md [hunk 5]
```diff
@@ -15,14 +15,42 @@ Build the site with:
...
+* `git merge <repo_where_the_changes_are>/master`; e.g. `git merge <friend_contributor>/master` or `git merge <your_github_user>/master`
+* `git log -p`. Check if the changes were properly merged.
+* git push upstream asf-site
```
Comment (swill) [inline README.md]: `git push upstream asf-site` (added backticks)

File: README.md [hunk 6]
```diff
@@ -15,14 +15,42 @@ Build the site with:
...
+* git push upstream master
```
Comment (swill) [inline README.md]: `git push upstream master` (added backticks)

**Comment by rafaelweingartner** [comment]
Changes suggested by @swill applied.

**Comment by rafaelweingartner** [comment]
Guys (@pdion891, @terbolous) any extra comment here?

**Comment by pdion891** [comment]
I haven't follow this whole thread but shouldn't we not include .html files in PR and master branch and only push them in the asf-site branch once approved ?  this would make master and PR cleaners.

**Comment by rafaelweingartner** [comment]
@pdion891, 
you have a double negation in the sentence, but I am reading as "we should not include *.html files in the PR. We should only generate and send them (HTML files) to `asf-site` branch after the PR is merged."

That is a good question. I only followed the pattern that has been used so far. I could remove the `content` from the master branch, though. Then, we could only maintain the `content` folder with HTMLs only on `asf-site` branch.

I do not mind working in one way or the other; It is up to you guys.

**Comment by pdion891** [comment]
sorry,  so yeah,   no html in the PR, only in asf-site when merge is aprouve, than html generated by the commiter merging the PR.  make sense?   since `content` folder already exist, no need to change this PR for this ;-)

**Comment by rafaelweingartner** [comment]
@pdion891, got your idea; but, for this PR do we keep with the current "working model, versioning the `content` folder in `master` branch" or you would like to change this?

**Comment by pdion891** [comment]
Let's change the model but not in this PR ;)

**Comment by swill** [comment]
So how would this work?  We would have a `.gitignore` tracked in `master` which would make the `content` folder untracked?  The `.gitignore` tracked in the `asf-site` branch would not include the `content` directory?

So then when you develop, it would not be as simple as a `merge` from `master` into `asf-site` anymore because the tracked files are not consistent between those branches anymore.

Would the flow be something like this then:
- Get the PR approved in `master`.
- Do a `merge` from `master` to `asf-site`.
- Run the `./build.sh` script in `asf-site`.
- Push code to `asf-site` branch.

Does that seem right?  We would have to do a `git rm` of the content on the `master` branch and then update the `.gitignore` tracked in that branch.

I don't think we need to worry about these changes in this PR though.

File: content/about.html [hunk 7]
```diff
@@ -79,7 +78,7 @@
                 <li><a tabindex="-1" href="contribute.html">Get Involved</a></li>
                 <li><a tabindex="-1" href="developers.html">Developers</a></li>
                 <li><a tabindex="-1" href="mailing-lists.html">Mailing Lists</a></li>
-                <li><a tabindex="-1" href="http://lanyrd.com/topics/apache-cloudstack/" target="_blank">Events &amp; Meetups <span class="glyphicon glyphicon-share-alt pull-right"></span></a></li>
```
Comment (swill) [inline content/about.html]: No one uses it though. It is just empty. Do we even have credentials somewhere to add to that?

Comment (rafaelweingartner) [inline content/about.html]: I was going to say exactly that. At least for me, it feels something that we stopped using and maintaining a long time ago. Could not we use the system/website of CCC for spread the word about user group meeting too? Then, we centralize information.

**Comment by rafaelweingartner** [comment]
@swill the procedure you described seems to be the way to go. However, as our discussion regarding the Jenkins idea, I am not convinced about the benefits of the change in procedures (this time I am the skeptical one). Also, it feels that the change would bring complications for merges (it would not be straightforward anymore).

**Comment by swill** [comment]
Yes @rafaelweingartner, that is also my concern.  It **probably** would be fine because the `.gitignore` would handle the problem cases with the merges (I hope), but it would make it less obvious to the end user what was going on for sure.

**Comment by rafaelweingartner** [comment]
I agree with you @swill 
So, if we intend to change this working flow, we will have a detailed discussion somewhere else.
Can I assume we ok to go here, and merge this PR?

BTW: there is a PR for the Cloudstack-WWW that is going to have a birthday this month (February 24), I think we should all celebrate it ;)

**Comment by swill** [comment]
Yes, I think we are good to go to merge this change.

**Comment by rafaelweingartner** [comment]
Merged; the good part is that I followed the new tutorial we developed, and it worked like a charm ;)

**Comment by swill** [comment]
Beauty. Good job. Thanks for doing that. :)

**Pull request merged**